packages/gatsby/src/utils/page-data.ts

```diff
@@ -140,6 +141,28 @@ export async function flush(): Promise<void> {
     // them, a page might not exist anymore щ（ﾟДﾟщ）
     // This is why we need this check
     if (page) {
+      if (
+        program?._?.[0] === `develop` &&
+        process.env.GATSBY_EXPERIMENTAL_GET_PAGE_DATA
+      ) {
+        // check if already did run query for this page
+        // with query-on-demand we might have pending page-data write due to
+        // changes in static queries assigned to page template, but we might not
+        // have query result for it
+        const query = queries.trackedQueries.get(page.path)
+        if (!query) {
+          // this should not happen ever
```

Comment:
This can potentially happen if the page is deleted as we track it separately in `deletedQueries`  😬

Reply:
Would https://github.com/gatsbyjs/gatsby/blob/6ab7f707161ac78f9581417b573bf46a2e6e1704/packages/gatsby/src/utils/page-data.ts#L137-L143 skip earlier because of that?